clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThread.java

```diff
@@ -273,80 +250,28 @@ private void closeInternal(final Duration timeout) {
         }
     }
 
+    private void sendUnsentRequests(final Timer timer) {
+        // Poll to ensure that request has been written to the socket. Wait until either the timer has expired or until
+        // all requests have received a response.
+        while (!networkClientDelegate.unsentRequests().isEmpty() && timer.notExpired()) {
```

Comment:
Closing with timeout 0 would mean we don't send any closing requests, right? I think we should `poll` nevertheless, so we should check the timer at the end.

I think if we'd use the normal poll loop as long as `timeout > 0`, this function may not need to check the timer anyway, since it's only used if the time ran out and there are still unsent requests.

Reply:
sorry, to reiterate on your comment, perhaps your suggestion is: if time has run out, we do client.poll(0) to try to send and process the request one last time. if the time hasn't run out and there are still request to be sent, we continue to poll until all requests are sent and timer runs out.  Is this what you meant?

Reply:
Well, if all requests are sent, I wouldn't timeout. But otherwise, yes.